scalableminds/webknossos
# When splitting a tree, the "original" tree should keep its name

When splitting one tree into many, previously, it was not clear which of the trees would keep the original tree name, id and timestamp. Upon request, I changed this so that always the original tree (the tree part that contains the initial node) will keep the original values.

I'm using a heuristic to identify the tree part that contains the initial node. The tree part that is linked to the split edge or node by the node with the lowest id, is assumed to be the tree part that contains the initial node. From my considerations this should almost always be the case. Of course we could do something more elaborate and traverse the tree parts to identify the node with the lowest timestamp, but that would be more complicated and would need more code. Any thoughts on this?

### URL of deployed dev instance (used for testing):
- https://splittreenames.webknossos.xyz

### Steps to test:
- Open a tracing and create a tree (let's call it the original tree) with some branches.
- Delete nodes in the original tree - the original tree should keep its id and name, the split trees should get new names and ids.
- Delete edges in the original tree (`Ctrl + Shift + Del`) - the original tree should keep its id and name, the split trees should get new names and ids.

### Issues:
- fixes #3082 

------
